src/ar_gateway_h.erl

```diff
@@ -0,0 +1,140 @@
+-module(ar_gateway_h).
+-behaviour(cowboy_handler).
+-export([init/2]).
+
+-include("ar.hrl").
+-define(BH_SEARCH_TIMEOUT, 5000).
```

Comment:
I think, we have decided to only have a global timeout.

Reply:
Considering I wrote the whole handler, I feel pretty confident that `ar_tx_search:get_tags_by_id` is the only call that can take some time before returning. Therefore, I don't see the point in having a global timeout, although I could add it if you think it still makes sense here.

Reply:
Ah, sorry, we do actually have a global timeout in the non-gateway server now.

Yes, it's probably not useful to have a global timeout in the gateway server at the moment. In general I think it is a way (although crappy) to have at least some backoff in case one of the endpoints is overloaded for whatever weird reason.